src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1969,6 +1970,30 @@ private void HandleVariableDeclaration(IVariableDeclarationOperation operation)
             {
                 ILocalSymbol localSymbol = declarator.Symbol;
 
+                // We skip constants in the control flow graph, as they're not actually involved in any control flow.
+                if (localSymbol.IsConst)
+                {
+                    continue;
+                }
+
+                // If the local is a static (possible in VB), then we create a semaphore for conditional execution of the initializer.
+                BasicBlock afterInitialization = null;
+                if (localSymbol.IsStatic && (declarator.Initializer != null || operation.Initializer != null))
+                {
+                    afterInitialization = new BasicBlock(BasicBlockKind.Block);
+                    LinkBlocks(CurrentBasicBlock, afterInitialization);
+
+                    ITypeSymbol booleanType = ((Operation)operation).SemanticModel.Compilation.GetSpecialType(SpecialType.System_Boolean);
+                    var initializationSemaphore = new StaticLocalInitializationSemaphoreOperation(localSymbol, localSymbol.DeclaringSyntaxReferences.Single().GetSyntax(), booleanType, constantValue: null, isImplicit: true);
+                    Operation.SetParentOperation(initializationSemaphore, null);
+
+                    var initializerBlock = new BasicBlock(BasicBlockKind.Block);
+                    LinkBlocks(CurrentBasicBlock, (initializationSemaphore, JumpIfTrue: true, RegularBranch(initializerBlock)));
```

Comment:
>CurrentBasicBlock [](start = 31, length = 17)

The code uses CurrentBasicBlock property twice with the intent to work with the same basic block. However, the property itself can allocate and return new blocks. I think the code would be clearer and more robust if the property was used only once and the result was cached in a local.

Reply:
With the new structure I'm only calling `CurrentBasicBlock` once, so this isn't necessary.

---
In reply to: [174860738](https://github.com/dotnet/roslyn/pull/25458#discussion_r174860738) [](ancestors = 174860738)